### DISCUSSION TITLE
Seeking Funding or Sponsorship for the Flet Project

### DISCUSSION THREAD

**Discussion by mouadessalim** [flet-dev/flet#1884] [General]
Hello Flet community,

I wanted to start a discussion regarding the potential for seeking funding or sponsorship for the Flet project. As we all know, Flet aims to bridge Google Flutter and Python, providing an exciting opportunity to enhance Flutter's appeal and expand its user base.

In 2023, the mobile development landscape is highly competitive, with React Native leading the way. However, the Flet project's unique approach of combining the strengths of Flutter and Python could give Flutter the boost it needs to stand out.

**Has the project considered seeking direct funding from Google or sponsorship from other organizations?**

Google may see value in supporting a project that strengthens its own product. They have a history of backing open-source initiatives, and the Flet project aligns with their mission of making technology accessible to all. Exploring this avenue could significantly accelerate the project's development.

Additionally, seeking sponsorship from tech companies or organizations interested in both Flutter and Python could be a viable option. Companies looking to expand their mobile app development capabilities may find the Flet project an attractive proposition.

**The Potential Impact :**

The idea of using Python to interact with Flutter is not only developer-friendly but also broadens Flutter's scope. It simplifies the learning curve and attracts a more diverse community of developers. This collaboration between Flutter and Python could rejuvenate Flutter's standing and ecosystem.

Let's discuss the possibilities, share insights, and explore potential avenues for funding or sponsorship. Your thoughts and ideas on this matter would be greatly appreciated. Together, we can accelerate the development of the Flet project and make it even more valuable to the Flutter community.

Looking forward to hearing your thoughts and suggestions !

@FeodorFitsner

**Comment by ndonkoHenri** [comment]
Just wish to mention that the Flet project is funded atm by AppVeyor, @FeodorFitsner's company.

**Comment by mouadessalim** [comment]
"Yes, I'm aware, but I'm talking about boosting the project on a large scale, maybe getting more people on board, or finishing the project so that it can grow faster, because the idea could appeal to Google. Using their technology to open the door to a large community could clearly revitalize the project because the massive mistake made by Google is creating a whole programming language (Dart) but only being able to use it for Flutter, in reality."

**Comment by FeodorFitsner** [comment]
Funding would be great and we definitely need to try that way. First of all, we need to gain more visibility in Google/Flutter ecosystem - any ideas?

Imho, looking from Google perspective Flet takes quite a tricky position: yes, it uses Flutter under the hood - that's good, but it does not directly promotes Dart and the way Flutter apps should be written, but instead it says "do apps in Python" - that's kind of not good, right? :) In other words, does Flet appeal to Flutter developers?

**Reply by ndonkoHenri** [reply]
> Imho, looking from Google perspective Flet takes quite a tricky position: yes, it uses Flutter under the hood - that's good, but it does not directly promotes Dart and the way Flutter apps should be written, but instead it says "do apps in Python" - that's kind of not good, right? :)

I equally see it this way. That won't really sound "greattt" to them.

**Reply by mouadessalim** [reply]
To boost visibility, we can consider the following technical strategies:

**Showcase Success with Case Studies:** Creating in-depth case studies demonstrating how Flet has been leveraged to tackle specific challenges or create unique applications. These case studies can be shared via GitHub READMEs, developer forums, and relevant tech publications.

**Engage with the Flutter Community:** Active collaboration with the Flutter community can involve organizing code-driven workshops, webinars, or contributing actively to Flutter-related open-source projects. This approach solidifies our standing as a committed player within the Flutter ecosystem.

**Develop Educational Content:** Crafting technical tutorials, blog posts, and video tutorials showcasing how to use Flet in practical development scenarios. This content can cater to Flutter developers and inspire them to explore our Python-powered solution.

Considering Google's perspective, you've raised a valid concern. It's true that the primary use of Dart is Flutter, and this might have initially appeared as a limiting factor. However, in reality, this could be the solution to rejuvenate Flutter. Currently, Dart is experiencing a decrease in demand as competitors intensify their efforts to gain ground. For instance, React Native offers more freedom in terms of animations and effects, but its main challenge lies in time consumption. Developing apps with React Native often takes more time due to the need to write everything from scratch. In contrast, Flutter's ease of use allows for more efficient and effective project development. It's precisely this niche that I envision Flutter occupying, where it excels in creating projects swiftly and efficiently.

The vision I see for Flutter's future, which could completely change the game, is to transform Flutter into an SDK that can be used with multiple languages: Python, Go, Java, and more. This approach aligns with Microsoft's strategy with .NET and has the potential to rejuvenate Flutter and make it a more versatile and widely adopted framework.

Let's continue discussing these technical insights and suggestions. Your input is invaluable in shaping the direction of the Flet project and its role in the evolving mobile development landscape.

@FeodorFitsner

**Reply by ndonkoHenri** [reply]
Nice points.

> Crafting technical tutorials, blog posts, and video tutorials showcasing how to use Flet in practical development scenarios.

Some of the community members have already started with these...

**Reply by plugeit** [reply]
But python is used by google in many of its services especially AI/ml so flet could be a way to integrate ai/ml with gui making it easy for developers to develop apps like ChatGPT etc and use the libraries of python like PyTorch, Scikit-learn, OpenCV etc easily which is not possible to do with dart. Python community is also way larger and way more active than dart's. 

[https://www.linkedin.com/pulse/how-does-google-use-python-its-technology-stack-muqadas-rasheed-/](url)
